src/mscorlib/src/System/Random.cs

```diff
@@ -114,11 +121,39 @@ public Random()
           return retVal;
       }
 
+
+      [ThreadStatic]
+      private static Random t_threadRandom;
+      private static Random s_globalRandom = new Random(GenerateGlobalSeed());
+
+      /*=====================================GenerateSeed=====================================
+      **Returns: An integer that can be used as seed values for consecutively
+                 creating lots of instances on the same thread within a short period of time.
+      ========================================================================================*/
+      private static int GenerateSeed() {
+          if (t_threadRandom == null) {
+              int seed;
+              lock (s_globalRandom) {
+                  seed = s_globalRandom.Next();
+              }
+              t_threadRandom = new Random(seed);
+          }
+          return t_threadRandom.Next();
+      }
+
+      /*==================================GenerateGlobalSeed====================================
+      **Action:  Creates a number to use as global seed.
+      **Returns: An integer that is safe to use as seed values for thread-local seed generators.
+      ==========================================================================================*/
+      private static int GenerateGlobalSeed() {
+          return Guid.NewGuid().GetHashCode();
```

Comment:
I think the TickCount is enough if we are not considering multiple processes. If someone really cares about creating not-correlated seeds then they should probably not depend on the default ctor anyway.

My concern with using the Guid is that it introduces a bunch of computation. 

I like the idea of xor-ing TickCount with some other entropy that is process specific. Process Id is an obvious choice, but that value is not readily available.

On the other hand, this particular API is supposed to provide a good default for the seed. I think we can achieve that by using the current approach and having the seed for the global random be the TickCount. 

If there are scenarios where the seed needs to be initialized differently for 2 processes that start withing 15ms of each other then I think you should not be using the default seed anyway and instead rely on a different approach for generating the seed machine wide.